backend/devices/bitbox02/keystore_simulator_test.go

```diff
@@ -412,12 +419,77 @@ func makeTx(t *testing.T, device *Device, recipient *maketx.OutputInfo) *btc.Pro
 		GetPrevTx: func(chainhash.Hash) (*wire.MsgTx, error) {
 			return prevTx, nil
 		},
+		GetAddress: func(scriptHashHex blockchain.ScriptHashHex) (*addresses.AccountAddress, error) {
+			for _, address := range addrsInDifferentAccount {
```

Comment:
This is not a faithful mock, as this callback in prod also returns addresses of the same account, right? So you should include all addresses here :thinking:

Reply:
Given that I consolidated two functions in a single one, I changed the mock, lmk what you think of  this one